src/poetry/console/commands/search.py

```diff
@@ -21,11 +21,14 @@ class SearchCommand(Command):
     ]
 
     def handle(self) -> int:
-        from poetry.repositories.pypi_repository import PyPiRepository
+        seen = set()
 
-        results = PyPiRepository().search(self.argument("tokens"))
+        for result in self.poetry.pool.search(self.argument("tokens")):
```

Comment:
It would be nice if we could show the repositories where the package was found, like

```
Repository 1
-------------
- package (version)

Repository 2
-------------
- package (version)
-  ... 
````
etc

Reply:
An option to choose repos to search in (among the available ones) would also be nice.

Reply:
I added the repo names to the output for now. Will add the repo/source flag in a later PR.